File: src/main/java/net/arvandor/numinoustreasury/area/Area.java
package net.arvandor.numinoustreasury.area;

import org.bukkit.Location;
import org.bukkit.World;

public final class Area {

    private final World world;
    private final Location minLocation;
    private final Location maxLocation;

    public Area(World world, Location location1, Location location2) {
        this.world = world;
        if (location1.getWorld() != world || location2.getWorld() != world) {
            throw new IllegalArgumentException("Both location worlds must match world");
        }
        this.minLocation = new Location(
                location1.getWorld(),
                Math.min(location1.getX(), location2.getX()),
                Math.min(location1.getY(), location2.getY()),
                Math.min(location1.getZ(), location2.getZ())
        );
        this.maxLocation = new Location(
                location1.getWorld(),
                Math.max(location1.getX(), location2.getX()),
                Math.max(location1.getY(), location2.getY()),
                Math.max(location1.getZ(), location2.getZ())
        );
    }

    public World getWorld() {
        return world;
    }

    public Location getMinLocation() {
        return minLocation;
    }

    public Location getMaxLocation() {
        return maxLocation;
    }
    public boolean contains(Location location) {
        return getWorld() == location.getWorld()
                && getMinLocation().getX() <= location.getX()
                && getMinLocation().getY() <= location.getY()
                && getMinLocation().getZ() <= location.getZ()
                && getMaxLocation().getX() >= location.getX()
                && getMaxLocation().getY() >= location.getY()
                && getMaxLocation().getZ() >= location.getZ();
    }
}

File: src/main/kotlin/net/arvandor/numinoustreasury/area/Area.kt
package net.arvandor.numinoustreasury.area

import org.bukkit.Location
import org.bukkit.World
import kotlin.math.max
import kotlin.math.min

class Area(val world: World, location1: Location, location2: Location) {
    val minLocation: Location
    val maxLocation: Location

    init {
        require(!(location1.world !== world || location2.world !== world)) { "Both location worlds must match world" }
        this.minLocation =
            Location(
                location1.world,
                min(location1.x, location2.x),
                min(location1.y, location2.y),
                min(location1.z, location2.z),
            )
        this.maxLocation =
            Location(
                location1.world,
                max(location1.x, location2.x),
                max(location1.y, location2.y),
                max(location1.z, location2.z),
            )
    }

    fun contains(location: Location): Boolean {
        return world === location.world &&
            minLocation.x <= location.x &&
            minLocation.y <= location.y &&
            minLocation.z <= location.z &&
            maxLocation.x + 1 > location.x &&
            maxLocation.y + 1 > location.y &&
            maxLocation.z + 1 > location.z
    }
}
